includes/class-simple-local-avatars.php

```diff
@@ -342,10 +342,13 @@ public function get_simple_local_avatar_url( $id_or_email, $size ) {
 		}
 
 		// check rating
-		$avatar_rating = get_user_meta( $user_id, $this->rating_key, true );
-		$site_rating   = get_option( 'avatar_rating' );
-		if ( ! empty( $avatar_rating ) && 'G' !== $avatar_rating && $site_rating ) {
-			$ratings              = array_keys( $this->avatar_ratings );
+		$avatar_rating      = get_user_meta( $user_id, $this->rating_key, true );
+		$site_rating        = get_option( 'avatar_rating' );
+		$all_avatar_ratings = ! empty( $this->avatar_ratings ) && is_array( $this->avatar_ratings )
+			? $this->avatar_ratings
+			: array();
+		if ( ! empty( $avatar_rating ) && 'G' !== $avatar_rating && $site_rating && $all_avatar_ratings ) {
```

Comment:
Do we need the `$all_avatar_ratings` check here? Unless I'm missing something, in looking at the code changes here, `$all_avatar_ratings` will either be an array of ratings or will be an empty array which in either case, this check will always return `true` I think.

Reply:
Yes, even I was not getting the warning reported by the user mentioned at this [issue](https://wordpress.org/support/topic/error-during-authorization-to-rocketchat/)

I've just added a check for the safer side just in case some other plugin conflicts with this plugin.

Reply:
I guess to be more clear, I do think this code is great:
```php
$all_avatar_ratings = ! empty( $this->avatar_ratings ) && is_array( $this->avatar_ratings )
	? $this->avatar_ratings
	: array();
```

My question is on the addition of `$all_avatar_ratings` in the `if` statement. It doesn't seem like that will ever do anything (that value will always be an `array` now and thus having it in that statement will never change things). Unless I'm missing something, I think we can remove that piece:
```suggestion
		if ( ! empty( $avatar_rating ) && 'G' !== $avatar_rating && $site_rating ) {
```

Reply:
Oh yes. Got it. Sorry, I misunderstood your query. Yes, we can remove that from `if`.